zulipterminal/ui_tools/views.py

```diff
@@ -1219,31 +1220,42 @@ def __init__(self, controller: Any, msg: Message, title: str,
                                                                  len(title))
         widgets = self.make_table_with_categories(msg_info, column_widths)
 
-        if message_links:
-            message_link_widgets = []
-            message_link_width = 0
-            for index, link in enumerate(message_links):
-                text, link_index, _ = message_links[link]
-                caption = '{}: {}\n{}'.format(link_index, text, link)
-                message_link_width = max(
-                    message_link_width,
-                    len(max(caption.split('\n'), key=len))
-                )
-
-                display_attr = None if index % 2 else 'popup_contrast'
-                message_link_widgets.append(
-                    MessageLinkButton(controller, caption, link, display_attr)
-                )
+        # slice_index = Number of labels before links + 1 newline
+        #               + 1 '<Any> Links' category label.
+        slice_index = len(msg_info[0][1]) + 2
 
-            # slice_index = Number of labels before message links + 1 newline
-            #               + 1 'Message Links' category label.
-            slice_index = len(msg_info[0][1]) + 2
-            widgets = (widgets[:slice_index] + message_link_widgets
+        if message_links:
+            msg_link_widgets, msg_link_width = self.set_links(message_links)
+            widgets = (widgets[:slice_index] + msg_link_widgets
                        + widgets[slice_index:])
-            popup_width = max(popup_width, message_link_width)
+            popup_width = max(popup_width, msg_link_width)
 
         super().__init__(controller, widgets, 'MSG_INFO', popup_width, title)
 
+    def set_links(
```

Comment:
Actually, it uses `self.controller`, but that could be passed as a parameter too. (Since we're not modifying the class instance anyways :+1:) I'll update this in the coming iteration. Thanks for observing it :sweat_smile: 

Re the name, yeah I'll be naming it something like `get_link_buttons()`.